### ISSUE TITLE
GedcomX Documents are 6x larger than equivalent Gedcoms

### ISSUE THREAD

**Issue by jralls** [FamilySearch/gedcomx#173]
As first noted by [Tamura Jones](http://www.tamurajones.net/GEDCOMXConverter.xhtml) and repeated in #134, GedcomX produces files which, even after compression, are several times larger than the Gedcom they're derived from using the [Gedcom5 Coversion Utility](https://github.com/FamilySearch/gedcom5-conversion). (Jones found the number to be 8x, I found 6x. That's likely due to differences in the Gedcoms used for the test and that I used a Mac and he used a Win32 system.

Some of that excess size is the need to recite all of the namespace URIs in every document. For example: 

``` xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?> 
<gxc:relationship xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
                            xmlns:foaf="http://xmlns.com/foaf/0.1/"
                            xmlns:contact="http://www.w3.org/2000/10/swap/pim/contact#"
                            xmlns:ns4="http://purl.org/dc/terms/"
                            xmlns:gx="http://gedcomx.org/" 
                            xmlns:gxc="http://gedcomx.org/conclusion/v1/" 
                            rdf:ID="F58-I147-I165">
    <rdf:type rdf:resource="http://gedcomx.org/ParentChild"/>
    <gxc:person1 rdf:resource="persons/I147"/>
    <gxc:person2 rdf:resource="persons/I165"/>
</gxc:relationship>
```

That's 545 characters, compared to 40 to describe a 2-person family in Gedcom5:

```
0 @F01@ FAM
1 HUSB @I001@
1 WIFE @I002@
```

That's 13.5X before compression.

If the namespace declarations are moved to a DTD called gedcomx.dtd and included in every GedcomX Zip:

``` xml
<!ELEMENT gxc:relationship>
<!ATTLIST gxc:relationship xmlns:rdf CDATA #FIXED "http://www.w3.org/1999/02/22-rdf-syntax-ns#">
<!ATTLIST gxc:relationship  xmlns:foaf CDATA #FIXED "http://xmlns.com/foaf/0.1/">
<!ATTLIST gxc:relationship xmlns:ns4 CDATA #FIXED "http://purl.org/dc/terms/">
<!ATTLIST gxc:relationship xmlns:gx CDATA #FIXED "http://gedcomx.org/">
<!ATTLIST gxc:relationship  xmlns:gxc CDATA #FIXED "http://gedcomx.org/conclusion/v1/">
<!ATTLIST gxc:relationship rdf:ID CDATA>
```

That's obviously only the fragment of the DTD needed to encode the namespaces for the relationship element; the actual DTD would be much larger.
Then our relationship element becomes:

``` xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<!DOCTYPE gnc:relationship SYSTEM "gedcomx.dtd">
<gxc:relationship rdf:ID="F58-I147-I165">
    <rdf:type rdf:resource="http://gedcomx.org/ParentChild"/>
    <gxc:person1 rdf:resource="persons/I147"/>
    <gxc:person2 rdf:resource="persons/I165"/>
</gxc:relationship>
```

Only 323 characters. Better, but still about 8x, and it carries the penalty that we have to use a validating parser. What might a minimal, RDF-free XML representation, where the XML is part of a single document and is written as tersely as possible, give us? Here's an example:

``` xml
<gxc:relationship gxc:reltype="ParentChild" id="F58-I147-I165" gxc:person1="persons/I147" gxc:person2="persons/I165"/>
```

118 characters, or 3x. Considering that XML is a lot more verbose than Gedcom5, not unreasonable. It's partly down to the extra length of the identifiers, and to general redundancy: Consider that all of the information in this element is encoded in the id! 
That's in part because the example I chose doesn't have any SourceReferences, Notes, or an Attribution.

Not a very good Gedcom, but perhaps less contrived that Jones's, which from the names are ones he generated with his Gedfan program and apparently which contain no actual data. 

So to summarize, the increased footprint has three sources: Lots of tiny documents with redundant namespace URIs, RDF, and the general verbosity of XML. That's balanced against using existing standards and the flexibility to reference data outside of the GedcomX file.

Is it the right tradeoff?
